### PR TITLE
Update PTKView to resignfirst responder when the textfields is valid

### PR DESCRIPTION
**Pull request by tsivilay**

### CODE REVIEW

File: PaymentKit/PTKView.m [hunk 1]
```diff
@@ -637,7 +640,7 @@ - (PTKTextField *)nextFirstResponder;
     if (self.firstInvalidField)
         return self.firstInvalidField;
 
-    return self.cardCVCField;
+    return nil;
```
Comment (ThibaultKlein) [inline PaymentKit/PTKView.m]: can you put a quick dev note in the code to explain why you return nil?

File: PaymentKit/PTKView.m [hunk 2]
```diff
@@ -576,6 +578,7 @@ - (void)checkValid
 
         if ([self.delegate respondsToSelector:@selector(paymentView:withCard:isValid:)]) {
             [self.delegate paymentView:self withCard:self.card isValid:YES];
+            [self.cardCVCField endEditing:YES];
```
Comment (Haud) [inline PaymentKit/PTKView.m]: Why do we only end editing if the delegate responds to that selector?

**Comment by Haud** [comment]
Can we add a property or option for this behavior? like `shouldResignFirstResponderOnValidInput`? This is useful, but I am not sure we should force this on everyone.

**Comment by ThibaultKlein** [comment]
Agreed @Haud but can we make this behavior the default then, and document the option?

**Comment by ppatriotis** [comment]
@ThibaultKlein does it possibly introduce backwards compatibility issues if it's default?

**Comment by ThibaultKlein** [comment]
@ppatriotis potentially actually, this pod was not made to be changed like that at the first place, might be better to keep the pod behavior as default and make this new behavior as an option @tsivilay

**Comment by Haud** [comment]
@ThibaultKlein @ppatriotis Given that the default functionality is to not resign first responder, the default should be to not resign with the option given to resign automatically as advertised in this PR. Making this the default will automatically break any current usage once updated.

File: PaymentKit/PTKView.m (hunk 1 above)
Comment (Haud) [inline PaymentKit/PTKView.m]: +1 @ThibaultKlein

**Comment by ppatriotis** [comment]
1st rule of introducing backwards compatibility issues: Don't.

**Comment by Haud** [comment]
@ppatriotis  +1

**Comment by ThibaultKlein** [comment]
@Haud @ppatriotis if we make these changes optional then we should be good, @tsivilay any thoughts?

**Comment by Haud** [comment]
@ThibaultKlein Look at my comment above : I am not sure you want `endEditing:` only if the `delegate` exists.

**Comment by tsivilay** [comment]
I push another version with a delegate which can be used to resign if textfields are valid

File: PaymentKit/PTKView.h [hunk 3]
```diff
@@ -20,6 +20,7 @@
 @optional
 - (void)paymentView:(PTKView *)paymentView withCard:(PTKCard *)card isValid:(BOOL)valid;
 - (BOOL)paymentView:(PTKView *)paymentView shouldAcceptCardType:(PTKCardType)type;
+- (BOOL)shouldResignFirstResponderOnValidInput;
```
Comment (ThibaultKlein) [inline PaymentKit/PTKView.h]: I think just a BOOL property would do the job here

Comment (tsivilay) [inline PaymentKit/PTKView.h]: Did it on my last commit

**Comment by ThibaultKlein** [comment]
@Haud I'm good with the PR, can you have a look and merge when you're good as well?

File: PaymentKit/PTKView.h [hunk 4]
```diff
@@ -42,5 +42,6 @@
 @property (nonatomic, strong) UIColor *customFontColor;
 @property (nonatomic, weak) id <PTKViewDelegate> delegate;
 @property (nonatomic, readwrite) PTKCard *card;
+@property (nonatomic, assign) BOOL shouldResignFirstResponderOnValidInput;
```
Comment (Haud) [inline PaymentKit/PTKView.h]: Can you just AppleDoc this property? Other than that, I'm good to go.

I'm goign to file an issue to document all of these properties.

**Comment by Haud** [comment]
@ThibaultKlein @tsivilay Just add an Appledoc comment to the property header and i'm good.

**Pull request merged**